ldap/servers/slapd/log.c

```diff
@@ -2803,6 +2819,44 @@ get_log_sev_name(int loglevel, char *sev_name)
     return "";
 }
 
+static void
+log_append_error_buffer(time_t tnl, LogBufferInfo *lbi, char *msg, size_t size, int locked)
+{
+    slapdFrontendConfig_t *slapdFrontendConfig = getFrontendConfig();
+    char *insert_point = NULL;
+
+    /* While holding the lock, we determine if there is space in the buffer for our payload,
+       and if we need to flush.
+     */
+    PR_Lock(lbi->lock);
```

Comment:
is the flag 'locked' related to that lock ? shouldn't we test the flag

Reply:
> is the flag 'locked' related to that lock ? shouldn't we test the flag

Actually in my attempt to reduce all the duplicate code, and there's a lot more, we need to pass that lock flag but it's really only used for the error log when it comes down through slapd_log_error_proc_internal().